src/queries/useGetUserGaugePositions.tsx

```diff
@@ -0,0 +1,55 @@
+import { useQuery } from "@tanstack/react-query";
+import { useAccount } from "wagmi";
+
+import { request, gql } from "graphql-request";
+
+const SUBGRAPH_URL =
+  "https://api.studio.thegraph.com/proxy/50162/smartgarden-optimism-gauges/version/latest";
+
+export interface IGaugePositions {
+  id: string;
+  pool_name: string;
+}
+
+async function getUserGaugePositions(safeAddress: string | undefined) {
+  try {
+    if (!safeAddress) throw new Error("No Safe Account");
+    const querySafeAddr = gql`
+      query UserGaugePositions {
+        gaugePositions(
+          where: {
+            user: "${safeAddress}"
+            balance_gt: "0"
+          }
+        ) {
+          gauge {
+            id
+            protocol
+            pool
+            pool_name
+          }
+        }
+      }
+    `;
```

Comment:
so didnt you want to retrieve this from the separate `user_active_stakes.graphql` file? i liked this idea, even if it is just for syntax and maintenance.

Reply:
true, i will like to keep it on that way, will update it and hopefully can add also the chain-awareness

Reply:
#33

Reply:
i extracted the gql query into its separate directory for the time being, see be555d955c8a0578953bd8b009426172ed1043de

then, i think we can spend some more time in #33 to explore the right structure/architecture and if its possible to auto-generate all this helpers out of the queries itself